### ISSUE TITLE
[cmake] Remove obsolete tools (Trac #2205)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2197]
The following tools are checked by cmake but don't do anything. Remove them so that people don't waste anytime installing them for icetray

- mpi
- jni
- hippodraw
- hepmc
- doxygen -- used by the docs build system but not needed in cmake)
- photonics -- in simulation extra tools
- pthread -- maybe pthread is needed but no projects add it to their USE_TOOLS
- mysql -- some jeb projects include it in USE_TOOLS, but i don't think they actually use it for anything

Migrated from https://code.icecube.wisc.edu/ticket/2205
```json
{
    "status": "closed", 
    "changetime": "2019-03-12T16:26:47", 
    "description": "The following tools are checked by cmake but don't do anything. Remove them so that people don't waste anytime installing them for icetray\n\n- mpi\n- jni\n- hippodraw\n- hepmc\n- doxygen -- used by the docs build system but not needed in cmake)\n- photonics -- in simulation extra tools\n- pthread -- maybe pthread is needed but no projects add it to their USE_TOOLS\n- mysql -- some jeb projects include it in USE_TOOLS, but i don't think they actually use it for anything\n\n", 
    "reporter": "kjmeagher", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1552408007024069", 
    "component": "cmake", 
    "summary": "[cmake] Remove obsolete tools", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2018-11-14T15:02:48", 
    "milestone": "Vernal Equinox 2019", 
    "owner": "nega", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20181128T16:02:47: 
* olivas changed owner from "" to "nega"
* olivas commented:

> We should keep jni because EHE still needs it.


* olivas changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20190312T16:26:47: 
* nega changed status from "assigned" to "closed"
* nega changed resolution from "" to "fixed"
* nega commented:

> In changeset 171930:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="171930"
> purge/move old or meta-project specific tools
> 
> mpi -- dropped
> jni -- already included in ehe, dropped
> hippodraw -- dropped, deleted
> hepmc -- dropped
> doxygen -- detection moved to docs, deleted
> photonics -- n/a
> pthreads -- every linux user should have this, required for jeb/pnf
> mysql -- n/a
> 
> closes: #2197
> ```
